src/ruby/puppetserver-lib/puppet/server/puppet_config.rb

```diff
@@ -42,6 +42,15 @@ def self.initialize_puppet(puppet_config)
 
     Puppet::ApplicationSupport.push_application_context(master_run_mode)
 
+    # Puppet's https machinery expects to find an object at
+    # `Puppet.lookup(:ssl_context)` which it will then use as an input
+    # to the Verifier, which is passed to the client. We ignore these
+    # values when passed to our https client and manage our SSLContext
+    # in a completely different way.
+    #
+    # See `Puppet::Network::HttpPool.connection`
+    Puppet.push_context({ssl_context: :unused})
```

Comment:
@pcarlisle this is the puppet-server-using-push_context thing I mentioned at the soiree.